tooling/acvm_cli/src/cli/execute_cmd.rs

```diff
@@ -34,12 +34,19 @@ pub(crate) struct ExecuteCommand {
     /// Set to print output witness to stdout
     #[clap(long, short, action)]
     print: bool,
+
+    /// Use pedantic ACVM solving, i.e. double-check some black-box function
+    /// assumptions when solving.
+    /// This is disabled by default.
```

Comment:
Just wondering why not enable it by default? Is it expected to be on for production runs?

Reply:
It's expected to decrease performance, especially if we add additional checks for e.g. valid public keys.

Reply:
Okay, so it's something to do once. Performance is why I am not bullish on global-like approaches, they seem to involve some kind of sync primitive.